force-app/main/default/classes/ElevateBatchServiceValidRequest.cls

```diff
@@ -27,63 +27,321 @@
  *     ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  *     POSSIBILITY OF SUCH DAMAGE.
  */
-@isTest
-public with sharing class ElevateBatchServiceValidRequest extends ElevateBatchService {
+@IsTest
+public class ElevateBatchServiceValidRequest extends ElevateBatchService {
+    ElevateBatchItemType batchItemType;
+    String paymentMethod;
 
-    public override ElevateAuthorizedGift addToElevateBatch(ElevateTokenizedGift tokenizedGift, String elevateBatchId) {
-        ElevateAuthorizedGiftResponse giftResponse = addElevateBatchSuccessMessage();
+    private String oneTimeResponse {
+        get {
+            if (paymentMethod == 'CARD') {
+                return oneTimeCardSuccessResponse();
+            } else {
+                return oneTimeACHSuccessResponse();
+            }
+        }
+        set;
+    }
+    private String commitmentResponse {
+        get {
+            if (paymentMethod == 'CARD') {
+                return commitmentCardSuccessResponse();
+            } else {
+                return commitmentACHSuccessResponse();
+            }
+        }
+        set;
+    }
+
+    public ElevateBatchServiceValidRequest(){}
+    public ElevateBatchServiceValidRequest(ElevateBatchItemType batchItemType, String paymentMethod){
+        this.batchItemType = batchItemType;
+        this.paymentMethod = paymentMethod;
+    }
 
-        return new ElevateAuthorizedGift(tokenizedGift, giftResponse);
+    public override ElevateBatchItem addToElevateBatch(ElevateCreateBatchItemRequestDTO createBatchItemRequestDTO, String elevateBatchId) {
+        ElevateBatchItemCreateResponse giftResponse;
+
+        if (batchItemType == ElevateBatchItemType.ONE_TIME) {
+            giftResponse = ElevateBatchItemCreateResponse.parse(oneTimeResponse);
+        } else {
+            giftResponse = ElevateBatchItemCreateResponse.parse(commitmentResponse);
+        }
+
+        return new ElevateBatchItem(giftResponse);
     }
 
     public override ElevateBatch createElevateBatch() {
-        ElevateBatchResponse elevateBatchResponse = createElevateBatchSuccessMessage();
+        ElevateBatchResponse elevateBatchResponse = ElevateBatchResponse.parse(createBatchSuccessResponse());
         return new ElevateBatch(elevateBatchResponse);
     }
 
-    public ElevateBatchResponse createElevateBatchSuccessMessage() {
-        return (ElevateBatchResponse)JSON.deserialize('{\n' +
-                '  "id": "test-id",\n' +
-                '  "status": "Open"\n' +
-                '}', ElevateBatchResponse.class);
+    private static String oneTimeCardSuccessResponse() {
```

Comment:
I know this is mock data, but can we indicate that somehow? I was confused at first since this isn't a Test Class.